src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -0,0 +1,200 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <stdint.h>
+#include <memory>
+#include <set>
+
+#include "gtest/gtest.h"
+#include "application_manager/application.h"
+#include "application_manager/application_impl.h"
+#include "application_manager/application_manager_impl.h"
+#include "application_manager/mock_application_manager_settings.h"
+#include "application_manager/mock_resumption_data.h"
+#include "application_manager/resumption/resume_ctrl_impl.h"
+#include "application_manager/test/include/application_manager/mock_message_helper.h"
+#include "connection_handler/mock_connection_handler.h"
+#include "hmi_message_handler/mock_hmi_message_handler.h"
+#include "policy/mock_policy_settings.h"
+#include "policy/usage_statistics/mock_statistics_manager.h"
+#include "protocol_handler/mock_session_observer.h"
+#include "utils/custom_string.h"
+#include "utils/file_system.h"
+#include "utils/lock.h"
+#include "utils/make_shared.h"
+#include "utils/push_log.h"
+
+namespace test {
+namespace components {
+namespace application_manager_test {
+
+namespace am = application_manager;
+namespace policy_test = test::components::policy_handler_test;
+namespace con_test = connection_handler_test;
+
+using testing::_;
+using ::testing::Mock;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::NiceMock;
+using ::testing::SetArgPointee;
+
+using namespace application_manager;
+
+namespace {
+const std::string kDirectoryName = "./test_storage";
+const uint32_t kTimeout = 10000u;
+sync_primitives::Lock state_lock_;
+sync_primitives::ConditionalVariable state_condition_;
+}  // namespace
+
+class ApplicationManagerImplTest : public ::testing::Test {
+ public:
+  ApplicationManagerImplTest()
+      : mock_storage_(
+            ::utils::MakeShared<NiceMock<resumption_test::MockResumptionData> >(
+                app_mngr_))
+      , mock_message_helper_(
+            application_manager::MockMessageHelper::message_helper_mock())
+      , app_id_(0u) {
+    logger::create_log_message_loop_thread();
+    Mock::VerifyAndClearExpectations(&mock_message_helper_);
+  }
+  ~ApplicationManagerImplTest() {
+    Mock::VerifyAndClearExpectations(&mock_message_helper_);
+  }
+
+ protected:
+  void SetUp() OVERRIDE {
+    CreateAppManager();
+
+    ON_CALL(mock_connection_handler_, GetDataOnSessionKey(_, _, _, _))
+        .WillByDefault(DoAll(SetArgPointee<3u>(app_id_), Return(0)));
+    ON_CALL(mock_connection_handler_, get_session_observer())
+        .WillByDefault(ReturnRef(mock_session_observer_));
+
+    app_manager_impl_->resume_controller().set_resumption_storage(
+        mock_storage_);
+    app_manager_impl_->set_connection_handler(&mock_connection_handler_);
+  }
+
+  void CreateAppManager() {
+    const uint8_t expectedThreadPoolSize = 2u;
```

Comment:
@okozlovlux Done in https://github.com/smartdevicelink/sdl_core/pull/1234/commits/817018f15bcc9a280711589e063f3a1a262e115b